src/component/autocomplete.js

```diff
@@ -212,14 +237,23 @@ export class AutoCompleteCustomElement {
    * Set the text in the input to that of the selected item and set the
    * selected item as the value. Then hide the results(dropdown)
    *
-   * @param {Object} [result] when defined uses the result instead of the
-   * this.selected value
+   * @param {Object} [result] when defined uses the result instead of the this.selected value
    */
   onSelect(result) {
-    this.value        = (arguments.length === 0) ? this.selected : result;
-    this.results      = [];
-    this.justSelected = true;
-    this.search       = this.label(this.value);
+    result             = (arguments.length === 0) ? this.selected : result;
+    this.justSelected  = true;
+    this.value         = this.label(result);
+    this.previousValue = this.value;
+
+    this.setFocus(false);
+
+    // Because of the event we're in, databinding updates don't get called. Just skip a beat. :)
```

Comment:
Aurelia has taskQueue which enables one to schedule a task. This also works but might be less idiomatic.